chromeshack_posts.js

```diff
@@ -2,30 +2,77 @@ ChromeShack =
+            mutationsList.forEach(function(mutation) {
+                if (mutation.type == "childList") {
+                    mutation.addedNodes.forEach(function (node) {
+                        var elem = node.parentNode;
+                        var source_id = elem.id;
+                        var scrollOpts = {behavior: "instant", block: "center", inline: "nearest"};
+
+                        // starts with "root", they probably refreshed the thread
+                        if (node && node.id.indexOf("root_") == 0)
+                        {
+                            ChromeShack.processFullPosts(elem);
+                            if (getSetting("enabled_scripts").contains("scroll_to_post")) {
+                                // scroll our post into the view window
+                                elem.scrollIntoView(scrollOpts);
+                            }
+                        }
+                        else if (source_id == "postbox")
+                        {
+                            console.log(elem);
+                            ChromeShack.processPostBox(elem);
+                            if (getSetting("enabled_scripts").contains("scroll_to_post")) {
+                                // scroll our post into the view window
+                                elem.scrollIntoView(scrollOpts);
+                            }
+                        }
+
+                        // starts with "item_", they probably clicked on a reply
+                        if (source_id.indexOf("item_") == 0)
+                        {
+                            // grab the id from the old node, since the new node doesn't contain the id
+                            var id = source_id.substr(5);
+                            ChromeShack.processPost(elem, id);
+                            if (getSetting("enabled_scripts").contains("scroll_to_post")) {
+                                // scroll our post into the view window
+                                elem.scrollIntoView(scrollOpts);
+                            }
+                        }
+                    })
+                }
+            })
+        });
+        observer.observe(document, { characterData: true, subtree: true, childList: true });
+
+        // DEPRECATED: start listening for new nodes (replies) being inserted
+        // document.addEventListener('DOMNodeInserted', function(e) {
+        //     var source_id = e.srcElement.id;
+
+        //     // starts with "root", they probably refreshed the thread
+        //     if (source_id && source_id.indexOf("root_") == 0)
+        //     {
+        //         console.log(e.srcElement);
+        //         ChromeShack.processFullPosts(e.srcElement);
+        //     }
+        //     else if (source_id == "postbox")
+        //     {
+        //         console.log(e.srcElement);
+        //         ChromeShack.processPostBox(e.srcElement);
+        //     }
```

Comment:
Any reason to keep this old code around?

Reply:
I wasn't completely confident in how I ported Mutation Events over to MutationObserver(). I left it in just in case side-by-side testing was necessary.

Reply:
I like your optimism about the level of testing that will happen here before merging ;)  I'd rather just have a clean PR and then compare the old release vs. the new PR.

Reply:
I understand. We're all busy and have responsibilities. I promise I'll try to be thorough in testing on both browsers so this can be cleanly merged with minimal fuss.